doc/specs/#1620 - Query Refinement for Disambiguation.md

```diff
@@ -0,0 +1,275 @@
+
+The UX can be improved by giving users an option to select the package instead of forcing the command to be re-run with the package identifier.
+This selection option can also be extended to the `upgrade`, `show` and `uninstall` commands.
+
+## Inspiration
+
+There are many similarly-named packages in winget. When users try to install any application, there is a high chance of a collision causing multiple suggestions in the returned table. It is an annoyance to copy the package identifier and re-run the command.
+
+## Solution Design
+
+When the query for `install`, `upgrade`, `show`, or `uninstall` cannot be disambiguated to a single package, the user should be given the opportunity to continuously refine their query.
+
+> Query refinement should not occur if `--diable-interactivity` is used or if interactivity has been disabled through settings. Query refinement is not required for COM callers.
+
+### Settings
+
+This shall be an **opt-in** only feature. The setting for this feature shall be placed uder the `interactivity` object. 
+
+In order to enable the query refinement feature -
+> Note: The setting could also be called `packageDisambiguation`
+```json
+"interactivity" : {
+   "refineAmbiguousQueries": true
+}
+```
+In order to disable the query refinement feature -
+```json
+"interactivity" : {
+   "refineAmbiguousQueries": false // This is the default behavior
+}
+```
+
+From the command line, users should be able to use `--refine-query` to override the user setting and show the query refinement, and `--no-refinement` to override the user setting and not show the query refinement.
+
+### Refinement Source
+The results from the query needing refinement should be placed into a temporary database to be used for further refinement. This database should be treated as a new temporary source to be queried. Upon completion of the command, this temporary database shall be removed. If the database is already existing and a new command is run, the database shall be overwritten.
+
+### Refinement queries
+Input to the package refinement should be treated as a query by default. This should match the behavior of the parent command. Each entry should also be assigned an index number which can be used to reference the package. The index number shall be treated as a fully qualified reference, and entry of an exact match to the number as a query shall take precedent.
+
+> Example: A user runs `winget install Example` and receives the result:
+> ```
+> Multiple packages found matching input criteria.
+> Number Name       Id                   Source
+> -----------------------------------------------
+>   1    Example 3  Example.Example3     msstore
+>   2    Example 2  Example.Example2     winget
+> ```
+> If the user were to input `'Example 2'`, it would be treated as `winget install -q 'Example 2'` using only the results in the table above as the source
+ 
+Users should be able to refine their query using any of the standard or shared flags. 
```

Comment:
Again, I would prefer that it be more of a straightforward property search without any need to evaluate complex inputs like this.